shared/src/helpers/match-by-router.cjsx

```diff
@@ -16,15 +17,16 @@ matchProps = (Router, props, parentParams) ->
     <props.render {...componentProps} params={params} />
   )
 
+
 matchByRouter = (Router, InvalidPage, displayName = 'RouterMatch') ->
   match = (props) ->
     return null unless props.routes
 
-    <span>
+    <Switch>
       {for route, i in props.routes
-        <Match key={i} {...matchProps(Router, route, props.params)} />}
-      <Miss component={InvalidPage} />
-    </span>
+        <Route key={i} {...matchProps(Router, route, props)} />}
+      <Route component={InvalidPage} />
+    </Switch>
   match.displayName = displayName
```

Comment:
new router doesn't have `Miss` instead you just use a route without a `path` which will match anything.